lib/model.js

```diff
@@ -782,7 +782,10 @@ class Model {
     if (!options.include) return;
 
     options.include = _(options.include)
-      .groupBy(include => `${include.model && include.model.name}-${include.as}`)
+      .groupBy(include => {
+        const modelName =  include.model ? include.model.name : include.association;
+        return `${modelName}-${include.as}`;
+      })
```

Comment:
`baseMerge` call `_conformOptions` which mutates options.include and `_uniqInclude` is called after which process the mutated options.include so at this moment, include.model is always defined and include.association is an instance of Association, or I missed something?